tests/tests/test_basic_integration.py

```diff
@@ -37,43 +36,39 @@ class TestBasicIntegration(MenderTesting):
     def test_double_update_rofs(self, standard_setup_one_rofs_client_bootstrapped):
         """Upgrade a device with two consecutive R/O images using different compression algorithms"""
 
-        mender_clients = standard_setup_one_rofs_client_bootstrapped.get_mender_clients()
-
-        if not env.host_string:
-            execute(self.test_double_update_rofs,
-                    standard_setup_one_rofs_client_bootstrapped,
-                    hosts=mender_clients)
-            return
+        mender_device = standard_setup_one_rofs_client_bootstrapped.device
 
         # Verify that partition is read-only as expected
-        run("mount | fgrep 'on / ' | fgrep '(ro,'")
+        mender_device.run("mount | fgrep 'on / ' | fgrep '(ro,'")
 
         host_ip = standard_setup_one_rofs_client_bootstrapped.get_virtual_network_host_ip()
-        update_image_successful(host_ip,
-                                install_image="mender-image-full-cmdline-rofs-%s.ext4" % conftest.machine_name,
-                                compression_type="gzip")
-        run("mount | fgrep 'on / ' | fgrep '(ro,'")
-
-        update_image_successful(host_ip,
-                                install_image="mender-image-full-cmdline-rofs-%s.ext4" % conftest.machine_name,
-                                compression_type="lzma")
-        run("mount | fgrep 'on / ' | fgrep '(ro,'")
-
+        update_image_successful(
```

Comment:
Idea:
Why not have this be a part of the device interface also? Like run above?
so:
```
mender_device.update_image_successful
```

I also think the Helper could be embedded in the device maybe? 

So that instead of `Helpers.foo.bar(device)`,
 we can do: `device.foo.bar`

Reply:
This is a nice idea. I moved only the `ssh_is_opened` but looking at `Helpers` once again I see that absolutely everything is device specific.

Unless the other reviewers disagree, I will move all of it!

Reply:
Also, I have a small request to make, and I feel now is probably a good time.

Could we change `update_image_successful` to simply `update_image`. Successful image updates is what we do for a living, right? The errors could be explicit instead.

Thoughts?

Reply:
I don't have an strong opinion, I will rename it. I am trying not to be too strict with these in the test framework, but I want to make you happy ;)

Reply:
> This is a nice idea. I moved only the `ssh_is_opened` but looking at `Helpers` once again I see that absolutely everything is device specific.

Not really everything. And also, I misunderstood your comment as was looking at `helpers.py` instead of `common_update.py`.

So I have moved many `helpers.py` methods to the `MenderDevice` class (see upcoming PR update) but nothing from `common_update.py`. The reason is that I don't want to make the `device.py` depend on MenderAPI module (basically because of it's global scope and the coupling with mender-artifact, etc).

Anyway, I think the changes I have made are nice and you will like it. And we can eventually make it even better after the MenderAPI refactor (QA-118).